percona/controller/pgbackup/testutils_test.go

```diff
@@ -0,0 +1,91 @@
+package pgbackup
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/percona/percona-postgresql-operator/internal/naming"
+	pNaming "github.com/percona/percona-postgresql-operator/percona/naming"
+	v2 "github.com/percona/percona-postgresql-operator/pkg/apis/pgv2.percona.com/v2"
+	"github.com/percona/percona-postgresql-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+)
+
+type fakeClient struct {
+	client.Client
+}
+
+var _ = client.Client(new(fakeClient))
+
+func (f *fakeClient) Patch(ctx context.Context, obj client.Object, patch client.Patch, options ...client.PatchOption) error {
+	err := f.Client.Patch(ctx, obj, patch, options...)
+	if !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if err := f.Create(ctx, obj); err != nil {
+		return err
+	}
+	return f.Client.Patch(ctx, obj, patch, options...)
+}
```

Comment:
Do we need this? By removing it nothing fails on the controller tests.

Reply:
https://github.com/percona/percona-postgresql-operator/pull/1133/commits/3af52428f930aae6b7e603e9f3c74fc53b94dd66